### ISSUE TITLE
Support for GPX1.1

### ISSUE THREAD

**Issue by fvezzoli** [tkrajina/gpxpy#20]
I read all GPX standard and i found the difference between version.
The first difference is the moving of metadata in a "metadata" child. The difference in the metadata child is:
- the author/mail in the 1.0 become the author in 1.1 with personType
- the mail is divided in id and domain attribute
- the url/urlname in 1.0 become linktype in 1.1 and add the possibility to add more link
- added the copyrigth
In the other part of GPX standard the only difference is the url → link modify.
I try to adjust the parsing code to add the possibility for parsing both version.
For creation gpx code start at end of parsing code modification.

**Comment by b-jazz** [comment]
When you are talking about 'aaa/ccc' for nested tags, that would just be for the object model, right? When you output a new gpx file with gpxpy (after doing whatever processing you do), it will still convert it into the original form (assuming it is untouched), right? I think that is very crucial and just want to make sure that would be the case.

Also, wouldn't putting a slash in the name cause problems since it is a valid, non-reserved character and you might already have an xml element with the name "aaa/ccc"?

**Comment by alkino** [comment]
What do you think of?
```
<extensions>
   <aaa>bbb</aaa>
   <aaa>ccc</aaa>
</extensions>
```
```
{'aaa': ['bbb', 'ccc']}
```
```
<extensions>
   <aaa>
      <ccc yyy="qqq">ddd</ccc>
   </aaa>
</extensions>
```
```
{'aaa': {'ccc': 'ddd'}} #don't know about 'yyy' :s
```

**Comment by jaap3** [comment]
My (admittingly naive) idea would be to allow users to register a handler for extensions. gpxpy would then just pass on the extension nodes to the handler responsible for that extension type.

**Comment by JaniszM** [comment]
Will this branch be merged into master? (https://github.com/tkrajina/gpxpy/tree/gpx-1.1) I would like to have support for 1.1 because of <speed> tag :)

**Comment by tkrajina** [comment]
@alkino Yes, you found the problem :) what with `yyy`? I was thinking of something like 

`<aaa>www</aaa>`
translates to
`{'aaa': 'www'}`

and

`<aaa><ccc yyy="qqq">ddd</ccc></aaa>`
translates to
`{'aaa': {'$yyy':'qqq', 'ccc': 'ddd'}}`

But then what to do with `something` here: `<aaa>something<ccc yyy="qqq">ddd</ccc></aaa>`? Maybe we can quietly ignore this scenario :)

Also note that in the XML the order of subelements is preserved, but translating it into a `dict` the order is gone (i.e. one more XML information is lost, this may or may not be a problem for somebody).

@JaniszM If you all guys agree I can merge it in master and release a new version immediately *but* with one important exception -- the `exception` API *will* change in the future. Because the way it is now it works only for the most trivial situations. Something like @jaap3 proposed might help, but at the moment I don't have time to finish this :(

Ignoring exceptions, I think the gpx-1.1 branch should be ready for merging.

**Comment by jaap3** [comment]
Sorry did you mean extensions instead of exceptions?

**Comment by JaniszM** [comment]
I don't need it right now, so If you can work on exception/extension it's ok for me to wait a bit :) It's not critical point right now to have this <speed> tag from <extensions>. So feel free to do what you guys think is right or you need :)

**Comment by tkrajina** [comment]
OK, gpx-1.1 is now merged in master and a new version (1.0) is deployed on pypi. Thanks you all all for your suggestions and feedback.

**Comment by markdall** [comment]
Oh boy I wish any of this still worked. :( Apparently the following is just unparsable:

`    <extensions>
     <gpxtpx:TrackPointExtension>
      <gpxtpx:hr>140</gpxtpx:hr>
     </gpxtpx:TrackPointExtension>
    </extensions>
`
